APITestingFramework/StepDefinitions/User/Delete.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Text.Json;
+using Core;
+using Features.GeneralSteps;
+using Models;
+using RestSharp;
+using TechTalk.SpecFlow;
+
+namespace Features.User.Delete
+{
+    [Binding]
+    [Scope(Feature = "User Deletion")]
+    public class DeleteStepDefinitions : CommonSteps
+    {
+        private readonly ScenarioContext _scenarioContext;
+        private readonly RestHelper client = new RestHelper("https://todo.ly/api");
+
+        public DeleteStepDefinitions(ScenarioContext scenarioContext) : base(scenarioContext)
+        {
+            _scenarioContext = scenarioContext;
+        }
+
+        [When(@"the user submits a DELETE request to the API endpoint")]
+        public void WhentheusersubmitsaDELETErequesttotheAPIendpoint()
+        {
+            var url = "user/0.json";
+            client.AddDefaultHeader("Authorization", _scenarioContext["Authorization"].ToString()!);
+            client.AddDefaultHeader("Accept", "*/*");
+            _scenarioContext["Response"] = client.Delete(url);
+        }
+
+        [Then(
+            @"the API should return a (.*) status code and the user should be removed from the database"
+        )]
+        public void ThentheAPIshouldreturnastatuscodeandtheusershouldberemovedfromthedatabase(
+            int args1
+        )
+        {
+            var response = (RestResponse)_scenarioContext["Response"];
+
+            Assert.True(response.IsSuccessful);
+            Assert.Equal("OK", response.StatusCode.ToString());
+            var user = JsonSerializer.Deserialize<UserPayloadModel>(response.Content!);
+            Assert.IsType<UserPayloadModel>(user);
```

Comment:
also validate that the information that you are sending of the user is the correct i.e. name, email